docs/cqlengine/upgrade_guide.rst

```diff
@@ -1,7 +1,54 @@
 ========================
-Upgrade Guide
+Upgrade Guide 3.x to 4.x
 ========================
 
+The cqlengine query execution has been rewritten to support async queries. The synchronous
+execution is now underlying on the main asynchronous one. This can have some impacts if you
+are using a custom DMLQuery or override some model functions, like save, update, delete.
+You should now override the *asynchronous* function to make sure your code will be called. See
+the example below.
```

Comment:
umm. Could be an idea yeah. Not exactly sure how this will be done though.

Reply:
Ok. I have added a warning for the most common case: method overriding in Models. I decided to not add the same warning for DMLQuery and other *QuerySet classes since it was requiring more work, like adding a new MetaClass just for that + adding the logic to detect if the class is really one of the user. I  don't think it's really frequent users override the queryset or dmlquery. Let me know what you think.